### DISCUSSION TITLE
Aprender inglês antes de aprender a programar

### DISCUSSION THREAD

**Discussion by matheusml** [frontendbr/forum#838] [Discussão]
Galera bonita,

_tweetei_ recentemente [isso aqui](https://twitter.com/matheusml/status/937069250652659712):

> "sou júnior, qual a primeira linguagem que devo aprender?" Inglês.

Pra mim isso era óbvio, mas muita gente não concordou, então parece que não era tão óbvio assim :)

O que vocês acham?

**Comment by bernardodiasc** [comment]
sim! :smile:

**Comment by martinsbicudo** [comment]
Concordo. Eu sou Júnior, não tenho inglês fluente, apenas intermediário, e cada dia mais eu sinto falta. Hoje, claro, busco aprender pra compensar, mas não é a mesma coisa de já ter aprendido desde o começo.

**Comment by willianjusten** [comment]
Que engraçado, eu fiz um tweet um tempinho atrás praticamente igual. E a mesma pessoa que comentou ahuahua

Eu fiz um [post hoje falando um pouquinho do assunto](https://willianjusten.com.br/o-que-preciso-saber-para-trabalhar-no-exterior/).

O problema é que o pessoal lê isso e generaliza... Entendendo que se você não sabe inglês, você não é um bom programador...

O que acontece é que o inglês abre portas para um conhecimento muito maior. E por mais que digam "Mas não precisa de inglês para arrumar o primeiro emprego.", a questão não é essa.

Seu objetivo de vida é ser junior sempre e só arrumar esses empregos ou crescer e ter oportunidade de trabalhar até com gente de fora? Se a resposta for a segunda opção, então sim, o inglês precisa ser priorizado tanto quanto a linguagem de programação.

**Comment by martinsbicudo** [comment]
Realmente, pra começar não é um problema. Os obstáculos vem quando você quer crescer mais e sair do Júnior, por isso recomendo já começar a aprender/estudar desde o começo.

**Comment by lidimayra** [comment]
Acredito que o cerne de tanta discussão seja porque há duas ideias diferentes aí sendo confundidas.

São elas:
**a -** Um bom domínio de inglês é fundamental para que um desenvolvedor evolua na carreira e se torne um profissional completo.
**b -** O inglês é pré-requisito para qualquer pessoa ingressar na área de desenvolvimento de software.

Na minha concepção, eu tenho **a** como verdadeira e **b** como falsa. E a impressão que tenho é que, muitas vezes, quando uma discussão como esta é iniciada, as pessoas afirmam **b** sendo, na verdade, guiadas por **a**.

Particularmente, enxergo a afirmação **b** como bastante nociva, pois coloca-se aí uma barreira desnecessária para quem ainda está se encontrando profissionalmente. Discordo do pensamento que as pessoas precisam, necessariamente, aprender inglês **antes** de aprender a programar. Se tem vondade, quer aprender, mãos a obra, não é o domínio do idioma que vai impedir ninguém...

**Comment by marcelgsantos** [comment]
Oi pessoal, tudo bem?

Não concordo. Acho que ter que aprender inglês antes é mais um **obstáculo** para o aprendizado em programação, que já não é algo fácil. Isso aumenta a chance de um **programador iniciante** querer desistir e tornar a área de desenvolvimento um grupo pequeno, para poucos.

Já lidamos com várias **faltas de incentivo** para formar novos programadores. Coisas como "isso é coisa para nerd", "tem que ser bom em matemática" ou "mulher não sabe programar". Não podemos ter mais um impeditivo!

Nesse ponto, concordo muito com o @peas, não devemos adicionar barreira alguma para quem inicia em programação.

Já discutimos esse assunto em um encontro do FEMUG-SP e tivemos excelentes opiniões de ambos os lados. Lembro-me que me identifiquei e gostei bastante das opiniões da @keitoliveira e da @akfzambrana.

Sobre a afirmação "com inglês vamos ter acesso a materiais mais atuais" é algo que também discordo. Hoje contamos com uma **comunidade brasileira** muito grande e ativa. Existem projetos e aritgos que são traduzidos para o português em _questão de semanas_. Alguns, os mais importantes ou relevantes, em questão de dias. Na pior das hipóteses, tem Stack Overflow em português, artigos de desenvolvedores brasileiros ou tradutor online.

Entendam, não estou dizendo que não se deve aprender inglês, muito pelo contrário. O inglês é **importantíssimo**. Só acho que não deve ser requisito para aprender uma linguagem de programação. Aliás, o inglês deve ser aprendido para a vida. Ele abre inúmeras portas. Mas pode ser aprendido depois de se passar a ter domínio em alguma linguagem de programação ou ir aprendendo em paralelo.

É importante lembrar que vivemos em um **país desigual** e nem todo mundo tem a oportunidade de aprender inglês, mesmo com todos os recursos existentes.

@willianjusten Acho complicado dizer "problema é que o pessoal lê isso e generaliza" sem **estar no lugar** de um programador iniciante. A maioria deles não possuem experiência de vida ou profissional para saber como interpretar isso. Isso pode até ser claro para um programador experiente, mas para um programador iniciante que nem sabe muito bem qual linguagem ou framework aprender ou se faz faculdade ou não, pode ser mais uma decisão difícil. ;)

@lidimayra Excelente comentário. Penso bastante igual a você.

[]s

**Comment by peas** [comment]
Assim como o @marcelgsantos , eu não concordo. Acho inclusive complicado falar pra alguém que quer começar a programar que ele deve primeiro aprender inglês. Uma enorme e longa barreira nova acaba de ser colocada, em vez de mostrar que em apenas um dia ele pode escrever um "olá mundo" sem saber o que `alert` `puts` ou `printf` quer dizer em português. Eu aprendi assim. Gostaria de ter aprendido inglês antes? Claro que sim. Mas a pergunta é se é ou não necessário. Não é para esse inicio.  Em resumo, o que a @lidimayra respondeu :).

**Comment by ghost** [comment]
Acabei de ler o artigo do @willianjusten e sempre fico nessa. Estudo agora ou estudo depois de entrar na área. 
Então divido meu tempo em, 4 a 6 timer tomato para Front(Se não fosse o sono, seria mais.). 1:30h de almoço mais 2h de trânsito para aprender inglês. Até agora percebi que a leitura melhoro bastante.

**Comment by miguelzinhe** [comment]
Eu tô com o prof @marcelgsantos, não concordo com vocês não sobre o inglês ser fundamental. 

Acho que a galera tem que estourar essa bolha enorme que há na área de tecnologia que todos conseguem, todos tem as mesmas oportunidades e todos precisam ter o hábito de estudar milhares de coisas. O inglês é importante depois de um tempo, mas, e quem não tem o dinheiro e a oportunidade de estudar inglês e quer saber programar um PC ou um site?

Eu dei aula por 5 meses na ONG Recode, tinhamos parceiros por todos os lugares de SP e eu ensinei programação front-end pra uma galera que nem sabia pegar num mouse ou digitar qualquer coisa num bloco de notas. Minhas alunas não tinham a oportunidade, nem financeira e nem geográfica, de aprender inglês ou se dedicar ao inglês e à programação. O inglês é uma barreira enorme, mas ela pode ser quebrada depois de algum tempo pelas pessoas devs juniors e pelas oportunidades pós arrumar um emprego na área, que é algo SUPER complicado pra quem está começando.

Achei um pensamento elitista e apático ter que saber inglês pra saber programação. Espero que cêis repensem isso após os relatos aqui encontrados :)

**Comment by kvnol** [comment]
Concordo que saber inglês é sim importante, fundamental (não sou nem um pouco fluente), mas concordo, como o @marcelgsantos disse, também que o idioma não deve fazer outros estudos ficarem de lado. Nós, seres humanos, temos capacidade de aprender diversas coisas ao mesmo tempo e acho que uma coisa não impede a outra.

Se eu fosse começar a aprender inglês antes de começar a estudar programação, não estaria aqui agora comentando isso, pois a vida me faria gostar de outras coisas se eu não tivesse colocado logo a mão no código.

**Comment by TiagoDanin** [comment]
Eu vejo que é necessário pelo menos o básico, visto em que boa parte das funções padrões das linguagens de programação são escritos em Inglês.

**Comment by bernardodiasc** [comment]
Já ví muita gente que mal sabia a lingua nativa trabalhando com programação. Mas é claro que a qualidade do profissional não entra no mérito. Tem júniors e júniors, uns só querem ou precisam de um emprego, outros querem ser profissionais incríveis. As prioridades são diferentes pra cada tipo.

**Comment by doug2k1** [comment]
Inglês é importante para a vida, mas como já disseram, não deve ser barreira pra aprender outras coisas, como programar. 

O ideal talvez seja aprender junto, que seja 10% do seu tempo de estudo para inglês e 90% outras coisas (essa porcentagem vai da necessidade que cada um sentir), ou até mesmo fora do tempo de estudo, como jogar e assistir séries e filmes em inglês (vai vendo quando você consegue acompanhar sem ficar lendo as legendas, ou com legendas em inglês se a leitura já for boa).

**Comment by woliveiras** [comment]
Galera, entendo a importância do inglês e concordo com o pessoal que é **contra a afirmação** de inglês como primeira linguagem basicamente pelo seguinte:

Não existem vagas para juniores assim:

**requisitos**

* inglês

Os requisitos, normalmente, são:

**JavaScript, Java, PHP, Ruby, Python, C#**

E, sim, tem o idioma inglês como requisito, porém não é eliminatório para juniores.

A pessoa quer aprender uma linguagem e procurar emprego na área, só com inglês ela não consegue emprego algum.

Poderia então a pessoa aprender inglês e alguma linguagem de programação em paralelo? 

> Claro! \o/

Mas será que todo mundo tem tempo pra isso?

> Normalmente não. :(

Então a primeira linguagem que a pessoa deve aprender é uma que tenha mercado na região onde ela mora.

O inglês vai lhe fazer ter acesso a bons materiais? 

> Com certeza!

Porém é material para evoluir na carreira, não precisa ser o material para aprender do zero. Para aprender temos diversos canais de conteúdo free ou pagos, mas até aí não foi a **primeira linguagem** que a pessoa precisou aprender.

**Comment by bernardodiasc** [comment]
Acho que o mal entendido na discussão é pensar que a pessoa tem que ser fluente em ingles antes de querer aprender a programar, como se fosse fator impeditivo. Isso é bobagem de fato. Mas acredito que a pessoa júnior no ramo de informática tem que ser no minimo capaz de ler um texto em ingles e que seja com ajuda de tradutor, nao importa, o que importa é que seja relevante na priorização de estudos.

**Comment by diessica** [comment]
Português não é só a primeira linguagem que aprendi, como também a linguagem em que estudei para aprender a programar, e uma das minhas linguagens preferidas no mundo.

Eu acredito e sou testemunha de que não é necessário saber inglês para programar (e não são muito de vocês também?). Pode ser útil saber que outras linguagens existem, mas nem básico precisa. Ainda mais se você pensar em programação como conceito, como comunicação com computador, algo que pode ser feito sem nenhuma dessas linguagens que a gente geralmente pensa como linguagem de programação.

Óbvio, não vou entrar nesse mérito, pois sei do que a pergunta se tratou, muito embora ela não faça sentido. Afinal, fala de júnior, mas um júnior já sabe uma primeira linguagem, não? sei lá tô com sono

Então vou considerar que o júnior já programa, mas quer escolher uma linguagem para focar de verdade. Mesmo assim, ainda não precisa saber inglês. Existem linguagens de programação com comunidades gigantes e incríveis, distribuídas pelo mundo inteiro e com comunidades locais fortes (que **devem** ser encorajadas) o suficiente para ter conteúdo e meetups suficientes para, pelo menos, deixar um júnior feliz. Na época que eu aprendi eu usava fóruns de desenvolvimento web, tipo o iMasters, era onde eu trocava conhecimento e tirava dúvidas. E muito Google Translate envolvido também. Era o jeito que a gente fazia, e todo mundo da época aprendeu igual. Hoje tem ainda mais conteúdo, tem Stack Overflow em português, tem livros em português, temos ótimas referências na comunidade brasileira também...

Assim que você aprende mais, sai da superfície, o acesso a conteúdo fica cada vez mais limitado e você nota que depende cada vez mais de alguém fazendo o intermediário e traduzindo para você. Nesse caso, além da exposição que você já teve com o inglês através da exposição feita pela programação, o [FOMO](https://en.m.wikipedia.org/wiki/Fear_of_missing_out) te faz aprender inglês, junto de outras motivações que surgem com o tempo: sair do país e tal.

Mas não é barreira de entrada. E isso foi uma coisa triste de ler, porque já fazem barreira de entrada com programação ser difícil, coisa de nerd, só faz coisa chata escrevendo código na tela com a bunda na cadeira o dia inteiro etc, quando na verdade é muito mais divertido que isso (a parte da cadeira é vdd). Não tem pré-requisito nenhum, e é d+ que seja assim.

Uma resposta mais responsável para "qual linguagem aprendo primeiro??" seria uma linguagem que:
-  tenha uma comunidade forte. (Quando você é iniciante, você depende muito disso)
- esteja alinhada com seus objetivos: aprender um paradigma diferente? Crescer dinheiro em árvore? Trabalhar para uma empresa específica?

E não "inglês". Sei lá, programação é bem tri. Vamos deixar o ego de lado e continuar com a ideia de que qualquer um pode fazer, porque é bem isso aí mesmo.

(tô com sono e é 3 AM, mas queria muito escrever isso, amo esse fórum fui)

**Comment by mahenrique94** [comment]
Acho que inglês em nossa área é essencial, praticamente todas as documentações são escritas em inglês, então sem o conhecimento você não irá conseguir ler.

**Comment by suissa** [comment]
" Vamos deixar o ego de lado e continuar com a ideia de que qualquer um pode fazer, porque é bem isso aí mesmo."

Assim como qualquer um pode aprender inglês DURANTE a programação, problema não reside em não saber ANTES mas sim em NUNCA saber.

Não existe NENHUMA pessoa que progxrxamxa que seja FODA sem saber Inglês, é simples, se você não souber Inglês terá que depender da boa vontade dos brasileiros de traduzir documentações e artigos novíssimos.

Qualquer pessoa pode programar sem nem saber o Português direito, só ficará para trás do resto das pessoas que sabem Inglês.

Não existe NENHUMA desculpa para não se aprender Inglês nessa área ou vais programar em Portugol mesmo. LOL 💃

**Comment by pietrobelluno** [comment]
segue um link do que o CEO da Apple falou sobre: 
https://exame.abril.com.br/tecnologia/programar-e-mais-importante-do-que-falar-ingles-diz-ceo-da-apple/

**Comment by wilcorrea** [comment]
Então, mas é óbvio que é preciso ir aprendendo inglês enquanto programa. Não conheço nenhuma linguagem popular que não use inglês como base para seus comandos. Querendo ou não você vai ter que lidar com isso. 

Aprender inglês nesse caso também não é se matricular no Fisk ou usar duolingo, na verdade tem mais a ver com pegar uma doc, um dicionário e sacar o que está rolando ali. 

Se isso não faz sentido não de mais nada nessa vida.

**Comment by diessica** [comment]
Eu nem tô falando de ser foda, Suissa. Para começar, muita gente nem aborda suas carreiras com esse ego todo que você assume ser importante. Eu tô falando de construir coisas pra computadores e já era. De fazer um programa.  Precisa de inglês para isso? Não, e não.

Se você vir o tópico inicial da thread, fala sobre saber inglês ANTES, como primeira linguagem, e não em NUNCA saber. Nunca saber daí já é outro tópico, né? Que acho que já foi coberto o suficiente em podcasts, artigos, e até mesmo nesse fórum. Importância do inglês não tem debate, mas isso já está indo longe demais e sendo abordado de forma exagerada, não?

**Comment by danlaurent** [comment]
Minha opinião é que nada impede de estudar as duas coisas em paralelo, o inglês e a programação.
Acredito que pra começar não é necessário ter o inglês (claro que seria bom pelo menos o básico), mas pra sua evolução, é sim necessário, pq se não vc se limita muito em questão de estudo, dúvidas e etc.
Fora que estudando as duas em paralelo, uma vai ajudar na melhor compreensão da outra.

**Comment by diessica** [comment]
@danlaurent perfeito! Não tem segredo :)

Só não espalhar essa abordagem de "inglês primeiro" aí, porque não saber inglês não é barreira de entrada não.

**Comment by silvaMatheus** [comment]
Atualmente estou estudando os dois paralelamente, sempre procuro ler artigos e documentações em inglês para treinar então acabo estudando Programação e ao mesmo tempo treino minha leitura... Estou procurando novas maneiras de treinar minha conversação pois acho q esta muito fraca. Acho q ter um base básica do ingles ja ajuda pois com o tempo vc vai ir evoluindo. Sempre manter contato com o Idioma ajuda muito, ler, escutar, falar sao coisas q consequentemente fazem evouluir

**Comment by felipefialho** [comment]
É um ponto muito relevante, e inclusive coloquei [nesse artigo que escrevi](https://medium.com/@lfeh/15-coisas-que-faria-se-estivesse-iniciando-minha-carreira-como-desenvolvedor-9fdd0c17b473) como uma das coisas que gostaria de ter feito assim que iniciei minha carreira.

Eu concordo quando disseram não ser uma linguagem básica e sim um plus importante (na verdade **muito** importante). 

Penso que uma pessoa que não sabe inglês, vai estudar programação e de repente coloca essa barreira, pode atrapalhar o desenvolvimento. Mas não tenho dúvidas que uma pessoa que já saiba inglês e comece a estudar programação, vai ter mais facilidade, por uma série de motivos que foram listadas aqui.

E aprender inglês vai além de programação, é algo pra vida, pra facilitar sua comunicação com a maioria das pessoas do mundo, para ter dar mais chance de trabalhar fora do país (se assim você preferir) e acesso a milhares de conteúdos (de todas as disciplinas), que não temos em português.

___

Muito bom ver a @diessica por aqui, volte mais vezes :)

**Comment by willianjusten** [comment]
Entendi melhor os pontos abordados, principalmente no quesito que colocaram de que o pessoal pode entender como **mais uma barreira** para entrar na área e com isso se desistimular. E sim, eu concordo com o que foi dito pelo @marcelgsantos.

Mas eu continuo no ponto de *"O inglês precisa ser priorizado tanto quanto a linguagem de programação".* Estudar uma coisa não impede de estudar a outra, pelo contrário, as duas se complementam. Enquanto você está estudando programação, você pode pegar artigos em inglês e também acabar se acostumando e estudando inglês por ali, mesmo que superficialmente. É assim que se aprende uma língua, enfrentando ela e fazendo ela entrar mais e mais no seu cotidiano...

**Comment by omariosouto** [comment]
De que adianta saber inglês, se você não sabe nem que problema que você tá tendo na linguagem? A gente já tem conteúdo nacional bom o suficiente pra você iniciar, e pro avançado tem crescido cada dia mais. O inglês é só uma barreira que pode ser resolvida mediante a demanda da evolução da pessoa. Muitas coisas são cognatos e com prática a pessoa associa tudo eliminando um muro no começo.

**Comment by maugravena** [comment]
Discordo dessa afirmativa também, com o basicão (ou nem isso) do inglês a pessoa já consegue desenrolar na programação (if, else, while etc) e entender um pouco da documentação ou obter ajuda da comunidade brasileira mesmo.

Vejo um pouco de soberba nessa questão, provavelmente um brasileiro experiente que trabalha no exterior e da até palestra em inglês vai ser mais humilde e tratar o assunto com mais consciência que algumas pessoas que ficam batendo nessa tecla.

**Comment by laurindo** [comment]
Um coisa que aprendi e percebi é que linguagem/tecnologia se aprende muito rápido, enquanto que o inglês já é algo mais demorado, precisa de uma prática diária. Claro que aprender uma tecnologia ou linguagem tem que levar em consideração a prática, mas a curva de aprendizagem é bem menor do que você que não sabe inglês ou é muito básico e deseja começar a estudar pra valer e chegar em um nível capaz de ler e se comunicar facilmente.

Vi que lá fora tem recrutas procurando profissionais sem tanta experiência com tecnologia e exigindo mais o inglês(avançado ou fluente), por conta que é mais rápido adquirir experiência em uma nova tecnologia.

Pra aprender a programar não é necessário saber inglês inicialmente, agora se o foco é crescer profissionalmente, quebrar barreiras, conquistar novas oportunidades seja remota ou morando fora, pensando em ganhar mais ou não, seja pra ter uma qualidade de vida melhor, acredito que o inglês se torne prioridade em relação a querer aprender uma nova tecnologia da moda. 

Nesse quesito o foco no inglês é maior do que a linguagem/tecnologia.

Pra finalizar, eu acredito que o inglês deve ser priorizado desde a hora que você começa a aprender a programar. Hoje em dia está sendo requisito básico.

**Comment by bernardodiasc** [comment]
`if`, `else`, `while`, `for`, todas as propriedades do CSS, vários elementos do HTML, e muito mais, são palavras inglesas. Como alguem usa elas sem saber o que significa? Se na hora que a pessoa ta aprendendo a programar e alguem fala "`if` é `se`", pronto, já está aprendendo ingles...

**Comment by bernardodiasc** [comment]
tenta programar assim com isso `jos`, `muu`, `sillä aikaa`, `varten`... se nao souber o significado das palavras nao sei o que vai conseguir fazer...

**Comment by bernardodiasc** [comment]
se o idioma nao faz diferença isso `अगर`। `अन्य`। `जबकि`। `के लिये`। tambem nao seria impedimento se as linguagens de programação fossem nesse idioma. nao é pq `margin` é obviamente `margem` que a pessoa nao ta aprendendo o idioma.

`¯\_(ツ)_/¯`  <- esse idioma aqui é universal né...

**Comment by vicenthy** [comment]
É apenas simbologia, ela pode ser muito verbosa ou não. Você aprender palavras chaves é bem mais simples que aprender um novo idioma e uma nova cultura, são coisas bem diferentes. 

Concordo muito que a questão do inglês tem muito mais a ver com a bibliografia, mas existem ferramentas surgindo para facilitar essa questão e quebrar essa barreira. 
Ah outra coisa existem pessoas que se dão melhor com a sintaxe de programação em inglês (mesmo sem ter domínio do idioma) do que com Portugol

**Comment by eduardojmatos** [comment]
Que issue bacana!

Meu ponto sobre isso é: se precisar, estude inglês. Se não, o básico você vai "aprender por osmose" (como o @bernardodiasc disse, `if`, `else`, `while`, etc.).

Quando digo "se precisar", digo mais no caso de você quiser algo pra fora do país e etc.. Se não, aqui tem um mercado muito grande ainda pra nossa área. Não vejo essa necessidade tão urgente assim. Tem espaço pra muita gente que tá começando agora, sem a obrigatoriedade do inglês.

**Comment by lidimayra** [comment]
Exatamente!!
Quando dizemos que o domínio prévio do inglês não é fundamental, quer dizer que:
_"olha só, tá tudo bem se você não consegue compreender a ideia central de um texto em inglês, não é isso que vai te impedir de aprender programação. `if` significa `se` e `else` significa `senão`... posso usar esses caras para implementar essa lógica aqui ó (...)"_

Com a evolução do aprendizado, a necessidade do inglês vai se fazer cada vez mais presente e os objetivos podem ser alcançados de forma gradativa.

Afinal, a ideia do pensamento computacional não é justamente essa? Quebrar o problema em diversas partes pequenas para que o objetivo final possa ser alcançado?

Ninguém discute a importância do domínio de inglês para a carreira do desenvolvedor.

Mas quando alguém diz que isso é pré-requisito para o **aprendizado**, o que me parece é uma tentativa desesperada de supervalorização... galera insere níveis de complexidade desnecessários apenas porque isso exalta as próprias habilidades (mesmo que isso seja danoso para o aprendizado de outra pessoa, que trilhou um caminho diferente)


> se o idioma nao faz diferença isso अगर। अन्य। जबकि। के लिये। tambem nao seria impedimento se as linguagens de programação fossem nesse idioma.

Discordo dessa analogia @bernardodiasc. Note que estamos discutindo a necessidade de ter ou não domínio prévio do idioma, em nenhum momento dissemos que o aprendizado do idioma não é importante. Muito pelo contrário, incentivo que o idioma seja aprendido ao longo da jornada.

Se eu explico para um nativo da língua portuguesa que `if` significa `se` (como foi citado no seu exemplo mesmo), ele terá facilidade em compreender isso na próxima vez em que esbarrar com o termo. Se você me explicar o que significa `अगर`, ainda assim, eu terei muita dificuldade em compreendê-lo novamente, pois há uma camada a mais de complexidade nesse caso, dado que eu sequer reconheço os padrões envolvidos nessa grafia e  levarei muito tempo até aprender a diferenciar uma palavra da outra.

**Comment by isaquebc** [comment]
Meu ponto de vista é que para ser programador, não é preciso saber inglês, mas a empresa que você trabalha pode exigir isso, pois assim você consegue se virar melhor.

Logo, para ser um programador com conhecimento de um Junior, você não precisa de inglês, mas se você quiser uma vaga melhor, você vai sentir falta.

**Comment by didiraja** [comment]
Concordo com você mas o idioma não deve ser barreira para se iniciar em programação. Mas sem dúvidas, conhecê-lo te ajuda de ponta a ponta.

**Comment by isaquebc** [comment]
@didiraja Concordo, acho inclusive que se você aprende a codar, você aprende inglês mais facil, e se você aprende inglês primeiro, você aprende a codar mais fácil... Um não exclui o outro

**Comment by CinthiaQueiro** [comment]
Bom dia

Verdade! Uma coisa não impede a outra. Inclusive existe os cursos via
Code school que forçam muito o aprendizado em paralelo. Comprar livros da
área em inglês também é uma boa.

Em 4 de dez de 2017 11:08, "Daniel Laurentino" <notifications@github.com>
escreveu:
